lib/assert.js

```diff
@@ -548,3 +548,21 @@ function doesNotThrow(block, /*optional*/error, /*optional*/message) {
 }
 
 assert.ifError = function ifError(err) { if (err) throw err; };
+
+assert.isInstanceOf = function isInstanceOf(obj, constructor, message) {
+  if (typeof constructor !== 'function') {
+    throw new TypeError('Constructor must be a function');
+  }
+  if (isNullOrNonObj(obj)) {
+    fail(obj, constructor, message, 'instanceof', assert.isInstanceOf);
+  }
+  if (constructor === Array) {
```

Comment:
It doesn't make much sense to add a special check for `Array` as there aren't different execution environments on Node.js and this only checks `constructor.prototype`. I'll remove it.